src/add-return-info-csv-parser.js

```diff
@@ -0,0 +1,125 @@
+import { SphereClient } from 'sphere-node-sdk'
+import _ from 'underscore'
+import highland from 'highland'
+import JSONStream from 'JSONStream'
+import csv from 'csv-parser'
+import CONS from './constants'
+
+export default class AddReturnInfoCsvParser {
+  constructor (apiClientConfig, logger, config = {}) {
+    this.client = new SphereClient(
+      Object.assign(apiClientConfig, { user_agent: 'csv-parser-orders' })
+    )
+    this.logger = logger || {
+      error: process.stderr,
+      warn: process.stderr,
+      info: process.stdout,
+      verbose: process.stdout,
+    }
+
+    this.config = config
+    this.batchSize =
+      this.config.batchSize || CONS.standardOption.batchSize
+    this.delimiter =
+      this.config.delimiter || CONS.standardOption.delimiter
+    this.strictMode =
+      this.config.strictMode || CONS.standardOption.strictMode
+  }
+
+  parse (input, output) {
+    this.logger.verbose('Starting conversion')
+
+    let rowIndex = 1
+
+    highland(input)
+    .through(csv({
+      separator: this.delimiter,
+      strict: this.strictMode,
+    }))
+    .batch(this.batchSize)
+    .stopOnError(error => this.logger.error(error))
+    .doto((data) => {
+      this.logger.verbose(`Parsed row-${rowIndex}: ${JSON.stringify(data)}`)
+      rowIndex += 1
+    })
+    .flatMap(highland)
+    .flatMap(data => highland(this.processData(data, rowIndex)))
+    .stopOnError(error => this.logger.error(error))
+    .doto(data => this.logger.verbose(
+      `Converted row${rowIndex}: ${JSON.stringify(data)}`
+    ))
+    .reduce([], (memo, next) => {
+      /*
+        Reduce all orders to one order object
+        1. Groups all orders by the orderNumber,
+        2. Group all returnInfo of an order by the _returnId
+      */
+      const allOrders = memo
+      const curOrder = next
+      if (!allOrders.length)
+        allOrders.push(curOrder)
+      else {
+        let found1 = false
+        _.each(allOrders, (order) => {
```

Comment:
Add a test that makes sure multiple orders with the same order number in one CSV work. Now it's implicit.